docs/DeveloperGuide.md

```diff
@@ -1,34 +1,429 @@
+3. [Implementation](#3-implementation)  
+    3.1. [Save Feature](#31-save-feature)  
+    3.2. [Daily Route Planning Feature](#32-daily-route-planning-feature)  
+    3.3. [Finding The Shortest Route Feature](#33-finding-the-shortest-route-feature)  
+    3.4. [Custom Aliases Feature](#34-custom-aliases-feature)  
+    3.5. [History Feature](#35-history-feature)  
+4. [Appendix: Requirements](#4-appendix-requirements)  
+    4.1. [Product Scope](#41-product-scope)  
+    4.2. [User Stories](#42-user-stories)  
+    4.3. [Use Cases](#43-use-cases)  
+    4.4. [Non-functional Requirements](#44-non-functional-requirements)  
+    4.5. [Glossary](#45-glossary)
+5. [Appendix: Instructions for manual testing](#5-appendix-instructions-for-manual-testing)
+<!-- TOC -->
 
+---------------------------------------------------------------------------------------------
 
-## Product scope
-### Target user profile
+## *1. Introduction*  
+### 1.1. Overview
 
-{Describe the target user profile}
+NUSMaze is a Command Line Interface (CLI) based application that aims to simplify NUS Engineering students’ journey from one point to another within the Engineering and Computing faculties of NUS. The application allows users to find the best route from one block to another, add favourite locations, locate the nearest eatery and much more.
 
-### Value proposition
+The purpose of this developer guide is to aid any curious or interested contributor in developing NUSMaze further by providing more insight on how the features were implemented.
 
-{Describe the value proposition: what problem does it solve?}
+### 1.2. Setting up and getting started 
+Step 1. Ensure that Java 11 and IntelliJ Idea (or your preferred Java IDE) are installed in your computer.  
+Step 2. Fork the NUSMaze repo from [here](https://github.com/AY2021S2-CS2113T-T09-2/tp), and clone the fork into your computer.    
+Step 3. Configure the JDK in IntelliJ Idea to use JDK 11 by following instructions from [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).    
+Step 4. Import the project as a Gradle project.
+Step 5. If you had previously disabled the Gradle plugin, go to `File → Settings → Plugins` to re-enable them.  
+Step 6. Click on Import Project and select the build.gradle file.  
+Step 7. Navigate to the NUSMaze class via the path `src → main → java → seedu.duke → NUSMaze` and right click on it.  
+Step 8. Press run on the `Main()` method of NUSMaze.
 
-## User Stories
+If the set up process had been completed successfully, you should see the following message:  
+![Screenshot 2021-03-25 at 7 03 08 PM](https://user-images.githubusercontent.com/60348727/113017279-e14b9d00-91b1-11eb-8ec3-37c0c3f80475.png)
 
-|Version| As a ... | I want to ... | So that I can ...|
+---------------------------------------------------------------------------------------------
+
+## *2. Design* 
+### 2.1. Architecture
+![img.png](images/architecture.png)
+
```

Comment:
May be can consider to label all the figures use :) (eg. Figure 1. Architecture Diagram)